src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/scenes/ExtensionFieldSetsImpl.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/CommonPersistentData.h>
+#include <lib/support/Span.h>
+
+/**
+ * @brief Indicates the absence of a Scene table entry.
+ */
+#ifndef CHIP_CONFIG_SCENES_TABLE_NULL_INDEX
+#define CHIP_CONFIG_SCENES_TABLE_NULL_INDEX 0xFF
+#endif
+
+/**
+ * @brief The group identifier for the global scene.
+ */
+#ifndef CHIP_CONFIG_SCENES_GLOBAL_SCENE_GROUP_ID
+#define CHIP_CONFIG_SCENES_GLOBAL_SCENE_GROUP_ID 0x0000
+#endif
+
+/**
+ * @brief The scene identifier for the global scene.
+ */
+#ifndef CHIP_CONFIG_SCENES_GLOBAL_SCENE_SCENE_ID
+#define CHIP_CONFIG_SCENES_GLOBAL_SCENE_SCENE_ID 0x00
+#endif
+
+namespace chip {
+namespace scenes {
+
+typedef uint32_t TransitionTimeMs;
+typedef uint16_t SceneTransitionTime;
+typedef uint8_t TransitionTime100ms;
+
+constexpr SceneGroupID kGlobalGroupSceneId  = CHIP_CONFIG_SCENES_GLOBAL_SCENE_GROUP_ID;
+constexpr SceneIndex kUndefinedSceneIndex   = 0xff;
+constexpr SceneId kUndefinedSceneId         = CHIP_CONFIG_SCENES_TABLE_NULL_INDEX;
+static constexpr uint8_t kMaxScenePerFabric = CHIP_CONFIG_SCENES_MAX_PER_FABRIC;
+static constexpr uint8_t kMaxSceneHandlers  = CHIP_CONFIG_SCENES_MAX_CLUSTERS_PER_SCENES;

```

Comment:
I don't understand what `kMaxSceneHandlers` (which has to do with how many endpoints are around, right?) has to do with `CHIP_CONFIG_SCENES_MAX_CLUSTERS_PER_SCENES`....  Please document why these are the right values for things, at least.  But I don't think this is the right value.